optaplanner-core/src/test/java/org/optaplanner/core/impl/testdata/phase/event/TestdataSolutionSateRecorder.java

```diff
@@ -0,0 +1,37 @@
+package org.optaplanner.core.impl.testdata.phase.event;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.optaplanner.core.impl.phase.event.PhaseLifecycleListenerAdapter;
+import org.optaplanner.core.impl.phase.scope.AbstractStepScope;
+import org.optaplanner.core.impl.solver.scope.DefaultSolverScope;
+import org.optaplanner.core.impl.testdata.domain.comparable.TestdataEntityWithDifficultyComparator;
+import org.optaplanner.core.impl.testdata.domain.comparable.TestdataSolutionWithDifficultyComparatorEntity;
+
+public class TestdataSolutionSateRecorder extends PhaseLifecycleListenerAdapter<TestdataSolutionWithDifficultyComparatorEntity> {
```

Comment:
This doesn't belong in the "testdata" package because it's not a domain object.
Put it near the test (see later comment).